src/features/wallet/root/useActivityItem.tsx

```diff
@@ -0,0 +1,299 @@
+        case 'payment_v2':
+          return isSending(item) ? (
+            <SentHnt width={32} height={18} />
+          ) : (
+            <ReceivedHnt width={32} height={18} />
+          )
+        case 'assert_location_v1':
+          return <Location width={20} height={23} />
+        case 'rewards_v1':
+          return <Rewards width={26} height={26} />
+        case 'token_burn_v1':
+          return <Burn width={23} height={28} />
+        case 'transfer_hotspot_v1':
+        case 'add_gateway_v1':
+        default:
+          return <HotspotAdded width={20} height={20} />
+      }
+    },
+    [isSending],
+  )
+
+  const isFee = useCallback(
+    (item: AnyTransaction | PendingTransaction) => {
+      if (item instanceof PaymentV1 || item instanceof PaymentV2) {
+        return isSending(item)
+      }
+
+      if (item instanceof RewardsV1) {
+        return false
+      }
+
+      if (item instanceof TransferHotspotV1) {
+        return (item as TransferHotspotV1).seller === address
+      }
+
+      return true
+    },
+    [isSending, address],
+  )
+
+  const formatAmount = (
+    fee: boolean,
+    amount?: Balance<DataCredits | NetworkTokens> | number,
+  ) => {
+    if (!amount) return ''
+
+    if (typeof amount === 'number') {
+      if (amount === 0) return '0'
+      return `${fee ? '-' : '+'}${amount.toLocaleString()}`
+    }
+
+    if (amount?.floatBalance === 0) return amount.toString()
+
+    return `${fee ? '-' : '+'}${amount?.toString()}`
+  }
+
+  const amount = useCallback(
+    (item: AnyTransaction | PendingTransaction, convertToHNT = false) => {
+      if (item instanceof AddGatewayV1) {
+        if (!convertToHNT) return animalHash(item.gateway)
+
+        return formatAmount(true, feeToHNT(item.fee))
+      }
+      if (item instanceof TransferHotspotV1) {
+        if (!item.fee) return ''
+
+        if (convertToHNT) {
+          return formatAmount(!isSelling(item), feeToHNT(item.fee))
+        }
+        return formatAmount(isFee(item), item.fee)
+      }
+      if (item instanceof AssertLocationV1 || item instanceof TokenBurnV1) {
+        if (!item.fee) return ''
+
+        if (convertToHNT) {
+          return formatAmount(true, feeToHNT(item.fee))
+        }
+        return formatAmount(isFee(item), item.fee)
+      }
+      if (item instanceof RewardsV1) {
+        return formatAmount(isFee(item), item.totalAmount)
+      }
+      if (item instanceof PaymentV1) {
+        return formatAmount(isFee(item), item.amount)
+      }
+      if (item instanceof PaymentV2) {
+        return formatAmount(isFee(item), item.totalAmount)
+      }
+      const pendingTxn = item as PendingTransaction
+      if (pendingTxn.txn !== undefined) {
+        if (pendingTxn.txn.type === 'add_gateway_v1') {
+          return animalHash((pendingTxn.txn as AddGatewayV1).gateway)
+        }
+        return formatAmount(isFee(item), pendingTxn.txn.fee)
+      }
+      return item.fee?.toString() || item.amount?.toString() || ''
```

Comment:
Have some TS errors here. Seems like fee may be missing from the type. Can we just add a TS ignore if its expected?